### PR TITLE
IS31FL3741 driver fixup

### PR DESCRIPTION
**Pull request by elfmimi**
## Description

Fix up a few things about IS31FL3741 driver.
- Fix issue with data transfer of CS1_SW7 to CS18_SW7. https://github.com/qmk/qmk_firmware/pull/10519/commits/0e40e9e7097401ca01d682e0714af43178a3e838
- Fix issue with handling of scaling register buffer's dirty flag. https://github.com/qmk/qmk_firmware/pull/10519/commits/05731f157a33a0cc86ee9012b4fa312de016486c
- Remove unused extern declaration. https://github.com/qmk/qmk_firmware/pull/10519/commits/f792491a848b6ea2d0380f26071a9f023fe2ab87
- Compaction of struct is31_led utilizing bit fields. https://github.com/qmk/qmk_firmware/commit/46538d9cd153afd356bc45ab22137b7b88644b5e
## Types of Changes

- [ ] Core
- [x] Driver
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #9961
* https://github.com/qmk/qmk_firmware/issues/10423

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: drivers/issi/is31fl3741.h [hunk 1]
```diff
@@ -23,14 +23,20 @@
 #include <stdbool.h>
 
 typedef struct is31_led {
-    uint8_t  driver : 2;
+#if __AVR__
```
Comment (fauxpark) [inline drivers/issi/is31fl3741.h:26]: AVR can do bitfields too...

Comment (elfmimi) [reply drivers/issi/is31fl3741.h:26]: I know. But it will generate really inefficient code, which I don't think justifiable.

Comment (fauxpark) [reply drivers/issi/is31fl3741.h:26]: AVR is already pretty inefficient, this won't make much difference.

Comment (elfmimi) [reply drivers/issi/is31fl3741.h:26]: good point. if you say so, let's remove it and make it simple.

**Review by fauxpark — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Comment by elfmimi** [comment]
Thank you for reviewing, as always. @fauxpark @tzarc

**Pull request merged**